lib/fastlane/plugin/wpmreleasetoolkit/actions/common/checkout_and_pull.rb

```diff
@@ -0,0 +1,39 @@
+require 'fastlane/action'
+require_relative '../../helper/git_helper'
+module Fastlane
+  module Actions
+    class CheckoutAndPullAction < Action
+      def self.run(params)
+        branch_name = params[:branch_name]
+
+        Fastlane::Helper::GitHelper.checkout_and_pull(branch_name)
+      end
+
+      def self.description
+        'Checkout and pull the specified branch'
+      end
+
+      def self.return_value
+        'True if it succeeded switching and pulling, false if there was an error during the switch or pull.'
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :branch_name,
+                                       env_name: 'BRANCH_NAME_TO_CHECKOUT_AND_PULL',
+                                       description: 'The name of the branch to checkout and pull',
+                                       optional: false,
+                                       type: String),
```

Comment:
In https://github.com/wordpress-mobile/release-toolkit/blob/c7329289f6f00a78cdf9f1988e627390efcf3c43/lib/fastlane/plugin/wpmreleasetoolkit/helper/git_helper.rb#L55C7-L57C118, the param can be either a `string` or `hash` with this explanation: 

```
# @param [String,Hash] branch Name of the branch to pull.
      #        If you provide a Hash with a single key=>value pair, it will build the branch name as `"#{key}/#{value}"`,
      #        i.e. `checkout_and_pull(release: version)` is equivalent to `checkout_and_pull("release/#{version}")`.
```

For this action, I just went with `string` (partially because I didn't think Fastlane action options could be defined with multiple types). The `hash` scenario seems a bit complex versus just specifying a string? If the `hash` option is needed, we could add an additional option for the `hash` type, but I would vote for just passing a string.